tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -984,6 +996,14 @@ public void onUpdateContainerResourceError(ContainerId containerId, Throwable t)
     @Override
     public void onContainersCompleted(List<ContainerStatus> completedContainers) {
       LOG.info("Completed containers: " + completedContainers.size());
+      if (System.getenv(Constants.TEST_TASK_COMPLETION_NOTIFICATION_DELAYED) != null) {
```

Comment:
add some comments here.. this is a testing block

Reply:
Extracted into a separate method.